main.tf

```diff
@@ -296,28 +297,28 @@ resource "aws_route_table_association" "private" {
   count = "${var.create_vpc && length(var.private_subnets) > 0 ? length(var.private_subnets) : 0}"
 
   subnet_id      = "${element(aws_subnet.private.*.id, count.index)}"
-  route_table_id = "${element(aws_route_table.private.*.id, count.index)}"
+  route_table_id = "${element(aws_route_table.private.*.id, (var.single_nat_gateway ? 0 : count.index))}"
 }
 
 resource "aws_route_table_association" "database" {
   count = "${var.create_vpc && length(var.database_subnets) > 0 ? length(var.database_subnets) : 0}"
 
   subnet_id      = "${element(aws_subnet.database.*.id, count.index)}"
-  route_table_id = "${element(aws_route_table.private.*.id, count.index)}"
+  route_table_id = "${element(aws_route_table.private.*.id, (var.single_nat_gateway ? 0 : count.index))}"
```

Comment:
It does not feel right that `aws_route_table_association` related to `database` has anything to do with the value of `var.single_nat_gateway` even if it is right in this particular piece of code.

Reply:
I debated created a new variable, since yes I am overloading `single_nat_gateway`, but in the end I didn't see it adding value. And this approach simplified some of the logic so I could re-use the local.

Either way I guess. `single_private_route_table`?

Reply:
I suppose I could also just create a local named `single_private_route_table` and set it to `var.single_nat_gateway`...